apps/zk-explorer/containers/RelatedProjectsContainer/client/RelatedProjectsCarousel.tsx

```diff
@@ -0,0 +1,62 @@
+'use client';
+
+import { type FC, ComponentProps, useMemo } from 'react';
+import Link from 'next/link';
+import { Swiper, SwiperSlide } from 'swiper/react';
+import { Pagination } from 'swiper';
+import 'swiper/css';
+import 'swiper/css/pagination';
+
+import { ProjectCard } from '../../../components/ProjectCard/ProjectCard';
+import type { ProjectItem } from '../../../components/ProjectCard/types';
+
+const sharedSwiperProps = {
+  spaceBetween: 20,
+  pagination: {
+    clickable: true,
+  },
+  navigation: true,
+  modules: [Pagination],
+} satisfies ComponentProps<typeof Swiper>;
+
+const RelatedProjectsCarousel: FC<{ projects: ProjectItem[] }> = ({
+  projects,
+}) => {
+  const projectsCmp = useMemo(() => {
+    return projects.map((project, idx) => (
+      <SwiperSlide key={idx}>
+        <Link
+          href={`/project/${project.repositoryOwner}/${project.repositoryName}`}
```

Comment:
Should we consider making an util function to create project URLs given repository owner and name parameters, since we're using a specific path in multiple places, but in case we decide to change that path in the future? Something like this:

```ts
export function createProjectUrl(repoOwner: string, repoName: string): string {
  return `/project/${project.repositoryOwner}/${project.repositoryName}`
}
```

That way we have a centralized function we can easily change in the future

Reply:
great call @yurixander!